src/BloomExe/Edit/EditingModel.cs

```diff
@@ -512,7 +514,7 @@ public void SetupServerWithCurrentPageIframeContents()
 		{
 			_domForCurrentPage = _bookSelection.CurrentSelection.GetEditableHtmlDomForPage(_pageSelection.CurrentSelection);
 			XmlHtmlConverter.MakeXmlishTagsSafeForInterpretationAsHtml(_domForCurrentPage.RawDom);
-			_server.CurrentPageContent = TempFileUtils.CreateHtml5StringFromXml(_domForCurrentPage.RawDom);
+			_nameForCurrentPage = _server.SetupCurrentPage(_domForCurrentPage.RawDom);
```

Comment:
Having the server be stateful (or more stateful, if it already is) seems like going the wrong direction.  Instead, can we get the browser to statelessly declare what it needs as much as possible, using url paths and query parameters?